tests/e2e/volumeproperty_test.go

```diff
@@ -0,0 +1,175 @@
+// TearDownTest - Removes all the containers and deletes the volumes
+func (s *VolumePropVerificationSuite) TearDownTest(c *C) {
+	out, err := dockercli.RemoveAllContainers(s.dockerHostIP)
+	c.Assert(err, IsNil, Commentf(out))
+	out, err = ssh.InvokeCommand(s.dockerHostIP, s.dockerVolumeRemoveCmd)
+	c.Assert(err, IsNil, Commentf(out))
+}
+
+var _ = Suite(&VolumePropVerificationSuite{})
+
+/*
+Test steps:
+1. SSH to a vm and create volumes of size 100 mb and specified disk format ["thin", "zeroedthick", "eagerzeroedthick"]
+2. Do docker inspect on the volume get size, disk format and attached-to-vm field.
+   Expected value: {"100MB", "thin/zeroedthick/eagerzeroedthick", no-value} respectively.
+3. SSH to the esx and get capacity, disk format and attached-to-vm field for the volume using Admin cli
+   Expected value: {"100MB", "thin/zeroedthick/eagerzeroedthick", "detached"} respectively.
+4. Verifies values obtained from admin cli and docker cli.
+5. SSH to the vm and run a container and mount the volume created.
+6. SSH to the vm and esx and verify the attached-to-vm field for volume - both docker cli and admin cli values should be same.
+7. Again verify capacity and disk format using docker cli and admin cli to make sure things are fine after running the container.
+
+NOTE: Do steps 5 ,6 and 7 only for volume of 'thin' disk format
+*/
+
+func (s *VolumePropVerificationSuite) TestVolumeProperties(c *C) {
+	log.Printf("START: TestVolumeProperties")
+
+	// creating volumes of all disk formats
+	s.createVolumes(c)
+	for _, volumeName := range s.volumeNames {
+		// getting and verifying the admin cli and docker cli values
+		s.getPropertiesAndVerify(c, volumeName)
+		s.dockerVolumeRemoveCmd = s.dockerVolumeRemoveCmd + " " + volumeName
+	}
+
+	// attaching only thin volume and verifying its properties
+	out, err := dockercli.AttachVolume(s.dockerHostIP, s.volumeNames[0], s.containerName)
+	c.Assert(err, IsNil, Commentf("Failed to attach the volume [%s]", out))
+	s.vmAttached = true
+	s.getPropertiesAndVerify(c, s.volumeNames[0])
+
+	log.Printf("END: TestVolumeProperties")
+}
+
+// createVolumes - creates volumes of each format type
+func (s *VolumePropVerificationSuite) createVolumes(c *C) {
+	for i, formatType := range s.formatTypes {
+		s.containerName = inputparams.GetContainerNameWithTimeStamp(TestName)
+		s.volumeNames[i] = inputparams.GetVolumeNameWithTimeStamp(TestName)
+		out, err := ssh.InvokeCommand(s.dockerHostIP, dockerconst.CreateVolume+"--name="+s.volumeNames[i]+" -o diskformat="+formatType)
+		c.Assert(err, IsNil, Commentf(out))
+	}
+}
+
+// getPropertiesAndVerify - gets properties of a volume from esx and docker host
+func (s *VolumePropVerificationSuite) getPropertiesAndVerify(c *C, volumeName string) {
+	valuesAdminCli, err := verification.GetVolumePropertiesAdminCli(volumeName, s.esxIP)
+	c.Assert(err, IsNil, Commentf(valuesAdminCli))
+
+	valuesDockerCli, err := verification.GetVolumePropertiesDockerCli(volumeName, s.dockerHostIP)
```

Comment:
Only a suggestion and you can decide how difficult or easy it may be to do this. The verifyProperties() function is may be a bit complex. Is it possible instead for the GetVolumeProperties*() functions to return a map of values so that the verification is easy to do. That would be ideal as it makes verifyProperties() pretty easy.